lib/active_merchant/billing/gateways/nmi.rb

```diff
@@ -175,6 +176,35 @@ def add_payment_method(post, payment_method, options)
           post[:ccnumber] = payment_method.number
           post[:cvv] = payment_method.verification_value unless empty?(payment_method.verification_value)
           post[:ccexp] = exp_date(payment_method)
+          add_stored_credential(post, options)
+        end
+      end
+
+      def add_stored_credential(post, options)
+        return unless (stored_credential = options[:stored_credential])
+
+        if stored_credential[:initiator] == 'cardholder'
+          post[:initiated_by] = 'customer'
+        else
+          post[:initiated_by] = 'merchant'
+        end
+
+        # :reason_type, when provided, overrides anything previously set in
+        # post[:billing_method] (see `add_invoice` and the :recurring) option
+        case stored_credential[:reason_type]
+        when 'recurring'
+          post[:billing_method] = 'recurring'
+        when 'installment'
+          post[:billing_method] = 'installment'
+        when 'unscheduled'
+          post.delete(:billing_method)
+        end
```

Comment:
Verified with NMI integration support over the phone that for unscheduled stored credential transactions the billing_method need not be set (it's only valid values are 'recurring' or 'installment').